app/src/main/java/com/example/myapplication/SecondFragment2.java

```diff
@@ -71,12 +76,36 @@ public View onCreateView(
         return binding.getRoot();
     }
 
+    /*Suppressed due to server having no way to get the dynamic path
+        TODO: Add matchNumber to file name once added
+    */
+    @SuppressLint("SdCardPath")
+    public void writeFile() throws IOException {
+        new File("/sdcard/Documents/ScoutingData/").mkdirs();
+
+        File dataFile = new File("/sdcard/Documents/ScoutingData/data.txt");
+        dataFile.createNewFile();
+        PrintWriter pw = new PrintWriter(dataFile);
+        pw.print(UserModel.getMatchData().returnAllData());
+        pw.close();
+        if (Build.VERSION.SDK_INT >= Build.VERSION_CODES.O) {
+            Files.deleteIfExists(Paths.get("/sdcard/Documents/ScoutingData/lock.txt"));
```

Comment:
Do we need a lock file? Right now there is no locking being done in this project.

Reply:
Yes, the computer via adb will access the files and transfer them. The absence of lock.txt means that the computer needs to update the changes. The computer will create lock.txt when it is done.